pkg/virt-api/rest/subresource.go

```diff
@@ -962,6 +940,54 @@ func generateVMVolumeRequestPatch(vm *v1.VirtualMachine, volumeRequest *v1.Virtu
 	return patch, nil
 }
 
+func addAddVolumeRequests(vm *v1.VirtualMachine, volumeRequest *v1.VirtualMachineVolumeRequest, vmCopy *v1.VirtualMachine) error {
+	name := volumeRequest.AddVolumeOptions.Name
```

Comment:
I believe it would help to separate adding single Request into a separate function and use it here